File: common/src/main/java/com/andyanika/translator/common/interfaces/usecase/SelectLanguageUseCase.java
package com.andyanika.translator.common.interfaces.usecase;

import com.andyanika.translator.common.models.LanguageCode;

import io.reactivex.rxjava3.core.Completable;

public interface SelectLanguageUseCase {
    Completable setSrc(LanguageCode code);

    Completable setDst(LanguageCode code);

    Completable swap();
}

File: common/src/main/java/com/andyanika/translator/common/interfaces/usecase/SelectLanguageUseCase.kt
package com.andyanika.translator.common.interfaces.usecase

import com.andyanika.translator.common.models.LanguageCode
import io.reactivex.rxjava3.core.Completable

interface SelectLanguageUseCase {
    fun setSrc(code: LanguageCode): Completable
    fun setDst(code: LanguageCode): Completable
    suspend fun swap()
}
